tests/local_evento_service_simulation.php

```diff
@@ -0,0 +1,446 @@
+    $this->config = new stdClass();
+    $this->config->adsidprefix = "S-1-5-21-";
+    $this->config->adshibbolethsuffix = "@fh-htwchur.ch";
+  }
+
+
+
+  /**
+   * Doing a simple init Webservice call to open the connection
+   * @return boolean true if the request was successfully
+  */
+  public function init_call()
+  {
+        return true;
+  }
+
+    /**
+     * Obtains an event by the id-number
+     * @param string $number the evento event-number like "mod.bspEA2.HS16_BS.001"
+     * @return stdClass event object "EventoAnlass" definied in the wsdl
+     */
+    public function get_event_by_number($number)
+    {
+
+      if($number = 'mod.mmpAUKATE1.HS18_BS.001'){
+      $anlassLeitungRolle_object = (object)
+      [
+        "anlassLtgRolleAktiv"=> true,
+        "anlassLtgRolleBezeichnung"=> "Hauptleitung",
+        "anlassLtgRolleBezeichnungKrz"=> "Hauptleitung",
+        "anlassLtgRolleBezeichnungSort"=> "100",
+        "idAnlassLtgRolle"=> 2
+      ];
+
+      $personenAnmeldungStatus_object = (object)
+      [
+        'idStatus'=>20175,
+        'statusName'=> "pA.Abgeschlossen"
+      ];
+
+      $array_personenanmeldungen_object = (object)
+      [
+        "iDAnmeldung"=> 123822,
+        "iDPAStatus"=> 20175,
+        "idAnlass"=> 25490,
+        "idPerson"=> 117828,
+        "personenAnmeldungStatus"=>$personenAnmeldungStatus_object
+      ];
+
+      $personenStatus_object = (object)
+      [
+          'idStatus'=>30040,
+          'statusName'=> "ps.Aktiv"
+      ];
+
+      $anlassLtgPerson_object = (object)
+      [
+        'array_personenanmeldungen' => array($array_personenanmeldungen_object),
+        'idPerson' => 117828,
+        'idPersonStatus' => 30040,
+        'personAktiv' => true,
+        'personKorrIdPerson' => 117828,
+        'personNachname' => "Müller",
+        'personRechIdPerson' => 117828,
+        'personVorname' => "Peter",
+        'personeMail' => "peter.mueller@htwchur.ch",
+        'personenStatus' => $personenStatus_object
+      ];
+/*---HIER BESTEHT EIN PROBLEM------*/
+      $array_EventoAnlassLeitung_object = (object)
+      [
+        'anlassLeitungRolle'=> $anlassLeitungRolle_object,
+        "anlassLtgIdAnlass"=>25490,
+        "anlassLtgIdAnlassLtgRolle"=>2,
+        "anlassLtgIdPerson"=>117828,
+        'anlassLtgPerson' => $anlassLtgPerson_object
+      ];
+
+      $anlassStatus_object = (object)
+      [
+      //  'idStatus' => 10230,
+      ];
+
+      $anlassKategorie_object = (object)
+      [
+        'anlassKategorieAktiv' => true,
+        'idAnlassKategorie' => 1
+      ];
+
+      $returnObject = (object)
+      [
+        'anlassBezeichnung'=> "Audio- & Kameratechnik 1",
+        'anlassDatumBis'=> "2019-02-17T00:00:00.000+01:00",
+        'anlassDatumVon'=> "2018-09-17T00:00:00.000+02:00",
+        'anlassKategorie' => $anlassKategorie_object,
+        'anlassLeitungIdPerson' => 117828,
+        'anlassNummer' => "mod.mmpAUKATE1.HS18_BS.001",
```

Comment:
Gewisse Angaben werden, z. B. mod.mmpAUKATE1.HS18_BS.001, werden mehrmals verwendet.
Das deutet darauf hin, dass man besser eine interne Datenstruktur verwendet.

Ev. ein https://de.wikipedia.org/wiki/Erbauer_(Entwurfsmuster) verwendet, um die interne Datenstruktur aufzubauen.

Reply:
Ich persönlich bin kein Anhänger der Erzeugunv von stdClass-Objekten mittels (object).
Damit gehen die Vorteile der Objektorientierung verloren. Das gibt zwar ein bisschen mehr Code, ich würde aber zur besseren Verständlichkeit trotzdem Klassen aus den Objekten machen.

class evento_service_anlassLtgRolle {
	// @var boolean
	public $anlassLtgRolleAktiv;
	
	// @var string
	public $anlassLtgRolleBezeichnung;
	
	$anlassLtgRolleBezeichnungKrz
	
	$anlassLtgRolleBezeichnungSort

	$idAnlassLtgRolle
	
	public function __construct(boolean $anlassLtgRolleAktiv, usw.) {
		$this->anlassLtgRolleAktiv = $anlassLtgRolleAktiv;
		$this->anlassLtgRolleBezeichnung = $anlassLtgRolleBezeichnung;
		$this->anlassLtgRolleBezeichnungKrz = $anlassLtgRolleBezeichnungKrz;
		Usw.
	}
	
}

class evento_service_anlassKategorie {
	// @var boolean
	public $anlassKategorieAktiv = true;
	
	// @var int
	public $idAnlassKategorie;
	
	public function __construct(boolean $anlassKategorieAktiv, $idAnlassKategorie) {
		$this->anlassKategorieAktiv = $anlassKategorieAktiv;
		$this->idAnlassKategorie = $idAnlassKategorie;
	}
}

class event_service_event {
	// @var string
	public $anlassNummer
	
	Usw.
}

class event_service {
	
	// @var array
	public $anlassKategories = array();
	
	// @var array
	public $events = array();
	
	// return event with given number
	public function get_event_by_number($anlassNummer) {
		foreach ($events as $event) {
			if ($anlassNummer== $event->anlassNummer) {
							return $event;
			}
		}
		return null;
	}
}

class evento_service_builder {
	
	// @var 
	public $event_service;
	
	public function __construct() {
		$event_service = new event_service();
	}
	
	public addAnlassKategorie(boolean $anlassKategorieAktiv, $idAnlassKategorie) {
		$anlassKategorie = new evento_service_anlassKategorie(boolean $anlassKategorieAktiv, $idAnlassKategorie);
		$event_service->anlassKategories[] = $anlassKategorie;
		return $anlassKategorie;
	}
}


// Dann hier das Builder-Pattern:
// Also der spezifische Teil
$builder = new evento_service_builder();
$var anlassLtgRolle = $builder.addAnlassLtgRolle(true, "Hauptleitung", "Hauptleitung", "100", 2);
$var anlassKategorie = $builder.addAnlassKategorie("true", 1);
$event = $builder.addEvent("Audio- & Kameratechnik 1", "2019-02-17T00:00:00.000+01:00", "2018-09-17T00:00:00.000+02:00", );

$service = $builder->event_service;

$service->get_event_by_number("mod.mmpAUKATE1.HS18_BS.001");